docs/pages/pmd/userdocs/extending/writing_java_rules.md

```diff
@@ -0,0 +1,164 @@
+---
+title:  Writing a custom rule
+tags: [extending, userdocs]
+summary: "Learn how to write a custom rule for PMD"
+last_updated: July 3, 2016
+permalink: pmd_userdocs_extending_writing_java_rules.html
+author: Tom Copeland <tomcopeland@users.sourceforge.net>
+---
+
+{% jdoc_nspace :coremx core::lang.metrics %}
+{% jdoc_nspace :coreast core::lang.ast %}
+{% jdoc_nspace :jmx java::lang.java.metrics %}
+{% jdoc_nspace :jast java::lang.java.ast %}
+{% jdoc_nspace :jrule java::lang.java.rule %}
+
+{% include note.html content="TODO All that should be written in the Javadocs,
+not sure we even need a doc page. Would be simpler to maintain too" %}
```

Comment:
Honestly, I think, we need such docs - lots of examples actually. It's easier to learn from examples and a website is more accessible than javadocs - you would need to have a starting pointer anyway, so at least a minimalistic documentation is always needed IMHO...